src/fal/planner/executor.py

```diff
@@ -152,3 +158,10 @@ def create_futures(executor: ThreadPoolExecutor) -> List[FutureGroup]:
             futures = get_futures(future_groups)
 
     _show_failed_groups(scheduler)
+    return _exit_code(future_groups)
```

Comment:
Wouldn't it look much simpler if just do any(scheduler.filter_groups(Status.FAILURE)) to check if there is anything failed?
```py
    return int(any(scheduler.filter_groups(Status.FAILURE)))
 ```

Reply:
how do you know that `int(...)` will be non-zero when there is something?

Reply:
If we were doing `status |= ...` before, why is it not relevant here?

Reply:
just to simplify the changes in this file and make it easier to understand (compared to `reduce()`ing) though no strong opinions.